_posts/2021--1-28-transactions-none-thanks.md

```diff
@@ -0,0 +1,41 @@
+---
+layout: post
+header_image: /img/posts/transactions-none-thanks/header.jpg
+title: "Transaction? None, thanks"
+author: Mauro Servienti
+synopsis: "Queues are designed for reliability. I personally stress a lot about designing message processing to be as transactional as possible. Is there a use case for unreliable message processing?"
+tags:
+- Distributed Systems
+- Transactions
+- Long running
+- Sagas
+---
+
+A few years ago, I wrote about long-running jobs in ["The case of NServiceBus long running jobs: OCR Processing."](https://milestone.topics.it/2016/12/20/the-case-of-nservicebus-long-running-handlers-ocr-processing.html) Please, go and read it.
+
+Long-running jobs have not to be confused with long-running business transactions; they are both long-running but different. 
+
+> Confusing, isn't it? Our industry is extremely good at terms overloading.
+
+Let's start with the difference between jobs and business transactions. A job, in this context, is a step in a business transaction. For example, the business transaction is the payment process, and one of the jobs is to authorize the user's credit card and another one to charge it. As demonstrated in the linked article, there are cases in which, when using message-based architectures, jobs can be problematic: this is when the job lasts longer than the allowed transaction timeout. Transaction to not be confused with business transactions, overloaded terms again.
+As demonstrated in the mentioned article, NServiceBus sagas, and messages in general, are not a good fit for extended processing time scenarios. If the processing time exceeds the transaction timeout, message processing fails, and the NServiceBus retry logic will try it again; it'll fail again, and again, and again. And eventually will end up in the error queue where there's not much we can do about it.
```

Comment:
```suggestion
As demonstrated in the mentioned article, NServiceBus sagas, and messages in general, are not a good fit for extended processing time scenarios. If the processing time exceeds the transaction timeout, message processing fails, and the NServiceBus retry logic will try it again; it'll fail again, and again, and again. Eventually it will end up in the error queue where there's not much we can do about it.
```